airbyte-integrations/connectors/destination-vectara/destination_vectara/writer.py

```diff
@@ -0,0 +1,115 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import dpath.util
+import uuid
+
+from typing import Any, Dict, List, Mapping, Optional
+
+from airbyte_cdk.models import ConfiguredAirbyteCatalog, AirbyteRecordMessage, ConfiguredAirbyteStream
+from airbyte_cdk.models.airbyte_protocol import DestinationSyncMode
+from airbyte_cdk.utils.traced_exception import AirbyteTracedException, FailureType
+
+from destination_vectara.client import VectaraClient
+
+
+
+METADATA_STREAM_FIELD = "_ab_stream"
+
+class VectaraWriter:
+
+    write_buffer: List[Mapping[str, Any]] = []
+    flush_interval = 1000
+
+    def __init__(self, client: VectaraClient, text_fields: Optional[List[str]], metadata_fields: Optional[List[str]], catalog: ConfiguredAirbyteCatalog):
+        self.client = client
+        self.text_fields = text_fields
+        self.metadata_fields = metadata_fields
+        self.streams = {f"{stream.stream.namespace}_{stream.stream.name}": stream for stream in catalog.streams}
+        self.ids_to_delete: List[str] = []
+
+    def delete_streams_to_overwrite(self, catalog: ConfiguredAirbyteCatalog) -> None:
+        streams_to_overwrite = [
+            stream.stream.name for stream in catalog.streams if stream.destination_sync_mode == DestinationSyncMode.overwrite
+        ]
+        if len(streams_to_overwrite):
+            self.client.delete_doc_by_metadata(metadata_field_name=METADATA_STREAM_FIELD, metadata_field_values=streams_to_overwrite)
+
+    def _delete_documents_to_dedupe(self):
+        if len(self.ids_to_delete) > 0:
+            self.client.delete_docs_by_id(document_ids=self.ids_to_delete)
+
+    def queue_write_operation(self, record: AirbyteRecordMessage) -> None:
+        """Adds messages to the write queue and flushes if the buffer is full"""
+
+        stream_identifier = self._get_stream_id(record=record)
+        document_section = self._get_document_section(record=record)
+        document_metadata = self._get_document_metadata(record=record)
+        primary_key = self._get_record_primary_key(record=record)
+
+        if primary_key:
+            document_id = f"Stream_{stream_identifier}_Key_{primary_key}"
+            self.ids_to_delete.append(document_id)
```

Comment:
It should probably only add that to that array if the sync mode of the stream is actually append_dedupe, right?

Reply:
Yes. Is the value "append_dedup" though (no "e")?

Reply:
See my fix - is that how it should be checked?